pkg/operator/ceph/csi/spec.go

```diff
@@ -315,6 +318,17 @@ func startDrivers(clientset kubernetes.Interface, rookclientset rookclient.Inter
 		}
 	}
 
+	tp.ProvisionerReplicas = defaultProvisionerReplicas
+	nodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		logger.Errorf("failed to get nodes. Defaulting the number of replicas of provisioner pods to %d. %v",
+			defaultProvisionerReplicas, err)
+	}
+	n := uint8(len(nodes.Items))
+	if n < defaultProvisionerReplicas {
```

Comment:
To make it more obvious that this is really for the single node case, how about this?
```
if len(nodes.Items) == 1 {
  tp.ProvisionerReplicas = 1
}
```